api/provider/aws/environment_delete.go

```diff
@@ -53,6 +59,42 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 	return nil
 }
 
+func (e *EnvironmentProvider) checkEnvironmentDependencies(environmentID string) error {
+	deleteEntity := func(entity string) error {
+		tags, err := e.TagStore.SelectByType(entity)
+		if err != nil {
+			return err
+		}
+
+		dependentTags := tags.WithKey("environment_id").WithValue(environmentID)
+		if len(dependentTags) > 0 {
+			IDs := []string{}
+			for _, tag := range dependentTags {
+				IDs = append(IDs, tag.EntityID)
+			}
+
+			msg := "Cannot delete environment '%s' because it contains dependent %s: %s"
```

Comment:
[nit] should be `... contains dependent %ss: %s` -> `... contains dependent services`

Reply:
Should we do it based on count, or is that an overkill 🤔